main.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'sinatra'
+
+DB_FILE_NAME = 'db.json'
+
+before do
+  @app_title = 'メモアプリ'
+end
+
+helpers do
+  def h(text)
+    Rack::Utils.escape_html(text)
+  end
+end
+
+class MemoDB
+  def self.show_memo_lists
+    generate_memo_lists['memo_lists']
+  end
+
+  def self.create_new_memo(title, content)
+    latest_id = generate_memo_lists['latest_id'] += 1
+    memo_lists = generate_memo_lists['memo_lists']
+
+    new_memo = { latest_id => { 'title' => title, 'content' => content } }
+    updated_lists = memo_lists.push new_memo
```

Comment:
pushは破壊的メソッドなので、新しい変数に代入してもあまり意味がないですね。
どうしても非破壊的にしたいなら、

```ruby
updated_lists = [*memo_lists, new_memo]
```

とかですが、ここは単純に

```suggestion
    memo_lists.push new_memo
```

だけでいいかなと思います。